File: component-service/src/main/java/dev/yidafu/pan/component/convertor/ComponentConvertor.java
package dev.yidafu.pan.component.convertor;

import dev.yidafu.pan.component.domain.dto.SaveComponentDTO;
import dev.yidafu.pan.component.domain.dto.UpdateComponentDTO;
import dev.yidafu.pan.component.domain.entity.Component;
import dev.yidafu.pan.component.domain.vo.ComponentVO;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper(componentModel = "spring")
public interface ComponentConvertor {
    ComponentConvertor INSTANCE = Mappers.getMapper(ComponentConvertor.class);

    List<ComponentVO> to(List<Component> list);

    ComponentVO to(Component com);

    Component from(SaveComponentDTO dto);

    Component from(UpdateComponentDTO dto);
}

File: component-service/src/main/kotlin/dev/yidafu/pan/component/convertor/ComponentConvertor.kt
package dev.yidafu.pan.component.convertor

import dev.yidafu.pan.component.domain.dto.SaveComponentDTO
import dev.yidafu.pan.component.domain.dto.UpdateComponentDTO
import dev.yidafu.pan.component.domain.model.Component
import dev.yidafu.pan.component.domain.vo.ComponentVO
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
interface ComponentConvertor {
    fun to(list: List<Component>): List<ComponentVO>
    fun to(com: Component): ComponentVO
    fun from(dto: SaveComponentDTO): Component
    fun from(dto: UpdateComponentDTO): Component

    companion object {
        val INSTANCE: ComponentConvertor = Mappers.getMapper(ComponentConvertor::class.java)
    }
}